pkg/condition/tiflow_cluster_executor_condition.go

```diff
@@ -0,0 +1,276 @@
+		SetTrue(v1alpha1.ExecutorReadyChecked, ecm.GetClusterStatus(), metav1.Now())
+		SetTrue(v1alpha1.ExecutorPVCChecked, ecm.GetClusterStatus(), metav1.Now())
+		SetTrue(v1alpha1.ExecutorsInfoUpdatedChecked, ecm.GetClusterStatus(), metav1.Now())
+		return nil
+	}
+
+	klog.Info("verify executor condition")
+	ns := ecm.GetNamespace()
+	tcName := ecm.GetName()
+
+	klog.Info("verify executor statefulSet condition")
+	sts, err := ecm.verifyStatefulSet(ctx)
+	if err != nil {
+		return result.NotReadyErr{
+			Err: err,
+		}
+	}
+
+	// todo: need to verify this
+	if ecm.pvcCheck() {
+		SetTrue(v1alpha1.ExecutorPVCChecked, ecm.GetClusterStatus(), metav1.Now())
+	} else {
+		SetFalse(v1alpha1.ExecutorPVCChecked, ecm.GetClusterStatus(), metav1.Now())
+		return result.NotReadyErr{
+			Err: fmt.Errorf("executor [%s/%s] verify: pvc's status is abnormal", ns, tcName),
+		}
+	}
+
+	klog.Info("verify executor version condition")
+	if ecm.versionVerify() {
+		SetTrue(v1alpha1.ExecutorVersionChecked, ecm.GetClusterStatus(), metav1.Now())
+	} else {
+		SetFalse(v1alpha1.ExecutorVersionChecked, ecm.GetClusterStatus(), metav1.Now())
+		return result.NotReadyErr{
+			Err: fmt.Errorf("executor [%s/%s] verify: version are not up-to-date", ns, tcName),
+		}
+	}
+
+	// todo: need to handle failureMembers
+	klog.Info("verify executor replica condition")
+	if ecm.ExecutorStsDesiredReplicas() == ecm.ExecutorStsCurrentReplicas() {
+		SetTrue(v1alpha1.ExecutorReplicaChecked, ecm.GetClusterStatus(), metav1.Now())
+	} else {
+		SetFalse(v1alpha1.ExecutorReplicaChecked, ecm.GetClusterStatus(), metav1.Now())
+		return result.NotReadyErr{
+			Err: fmt.Errorf("executor [%s/%s] verify: actual is not equal to desired replicas ", ns, tcName),
+		}
+	}
+
+	// todo: need to check this
+	klog.Info("verify executor all ready condition")
+	if ecm.ExecutorStsDesiredReplicas() == ecm.ExecutorStsReadyReplicas() {
+		SetTrue(v1alpha1.ExecutorReadyChecked, ecm.GetClusterStatus(), metav1.Now())
+	} else {
+		SetFalse(v1alpha1.ExecutorReadyChecked, ecm.GetClusterStatus(), metav1.Now())
+		return result.NotReadyErr{
+			Err: fmt.Errorf("executor [%s/%s] verify: cluster are not reday", ns, tcName),
+		}
+	}
+
+	if ecm.leaderVerify() {
+		SetTrue(v1alpha1.LeaderChecked, ecm.GetClusterStatus(), metav1.Now())
+	} else {
+		SetFalse(v1alpha1.LeaderChecked, ecm.GetClusterStatus(), metav1.Now())
+		return result.SyncStatusErr{
+			Err: fmt.Errorf("executor [%s/%s] verify: can not get ledaer from master cluster", ns, tcName),
+		}
+	}
+
+	if err := ecm.Update(ctx, sts); err != nil {
+		SetFalse(v1alpha1.ExecutorsInfoUpdatedChecked, ecm.GetClusterStatus(), metav1.Now())
+		return result.SyncStatusErr{
+			Err: fmt.Errorf("executor [%s/%s] verify: information of executor cluster update failed", ns, tcName),
+		}
+	}
+	SetTrue(v1alpha1.ExecutorsInfoUpdatedChecked, ecm.GetClusterStatus(), metav1.Now())
+
+	if ecm.ExecutorStsDesiredReplicas() == ecm.ExecutorAllActualMembers() {
```

Comment:
`ExecutorAllActualMembers` includes all peer members (like heterogeneous cluster). Maybe use `ExecutorActualMembers` here?

Reply:
When testing, I found that when the  executor cluster is expanded in same namespace, it will also be added to `peermMembers`, is this the expected behavior?